Core/Src/DynamicControls/lateral_control.c

```diff
@@ -10,73 +10,65 @@
 #include "DynamicControls/PID.h"
 #include "cmsis_os.h"
 #include "math.h"
+#include "speed_calc.h"
+#include "global_variables.h"
+#include "constants.h"
 
-extern volatile float g_wheel_speed[4];
-extern volatile uint16_t steering_wheel;
-extern volatile uint8_t internal_wheel;
-extern volatile uint16_t gyro_yaw; // ainda nao existe
-extern const float ku;
-extern const float wheelbase;
 extern PID_t pid_lateral;
 
-lateral_t lateral_control() {
+lateral_t lateral_control(volatile float g_wheel_speed[4], volatile uint16_t *steering_wheel, volatile uint8_t *internal_wheel, volatile uint16_t *gyro_yaw) {
```

Comment:
os parâmetros não precisam ser voláteis. Dá uma olhadinha aq o q significa uma variável volátil: https://www.geeksforgeeks.org/understanding-volatile-qualifier-in-c/

Reply:
Entendi, obrigada!

Reply:
Lembrei pq eu tinha feito isso. Tirando o "volatile" dos parametros, eu recebo o seguinte erro:
warning: passing argument 1 of 'lateral_control' discards 'volatile' qualifier from pointer target type [-Wdiscarded-qualifiers]

Reply:
esse warning aparece onde? talvez não tenha problema, dependendo de onde apareceu (lembrando q isso vai receber a partir de agora aquelas variáveis globais de queue)

Reply:
Aparece na torque manager, quando eu chamo a função.